pkg/providers/instancetype/types.go

```diff
@@ -206,23 +206,29 @@ func memory(ctx context.Context, info *ec2.InstanceTypeInfo) *resource.Quantity
 	return mem
 }
 
-// Setting ephemeral-storage to be either the default value or what is defined in blockDeviceMappings
-func ephemeralStorage(amiFamily amifamily.AMIFamily, blockDeviceMappings []*v1beta1.BlockDeviceMapping) *resource.Quantity {
-	if len(blockDeviceMappings) != 0 {
+// Setting ephemeral-storage to be either the default value, what is defined in blockDeviceMappings, or the combined size of local store volumes.
+func ephemeralStorage(info *ec2.InstanceTypeInfo, amiFamily amifamily.AMIFamily, nodeClass *v1beta1.EC2NodeClass) *resource.Quantity {
+	// If local store disks have been configured for node ephemeral-storage, use the total size of the disks.
+	if lo.FromPtr(nodeClass.Spec.InstanceStoreConfiguration) == v1beta1.MountNodeEphemeralStorage {
```

Comment:
Checking this assumption: If a user specifies this value with an AMIFamily that doesn't support this option, we just assume that a user has configured their userData correctly to automatically RAID0 these stores. I think this makes sense; just double-checking my understanding

Reply:
That sounds correct. Though it could get tricky if/when a currently-unsupported AMIFamily starts supporting this in the future. _Maybe_ that warrants another policy option for using the instance-store size here without the explicit bootstrap arg?